### PR TITLE
chore: add bazel test to CI

### LINKED ISSUE #236: chore: CI should test with bazel
In order to validate that the bazel build still works, CI should also test via `bazel test...`

### PR DESCRIPTION
**Pull request by noahdietz**
Runs `bazel test` as part of CI

Fixes #236

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -61,6 +65,27 @@ jobs:
             mv googleapis-master googleapis
             export PATH=$PATH:$(pwd)/protobuf/bin
             GOOGLEAPIS=googleapis OUT=$GOPATH/src ./test.sh
+  bazel-test:
+    environment:
+      BAZEL_VERSION: 1.2.1
```
Comment (vam-google) [inline .circleci/config.yml]: Please target bazel version 2+ (it is the newest verson, googleapis CI is build using this bazel version)

File: .circleci/config.yml [hunk 2]
```diff
@@ -61,6 +65,27 @@ jobs:
...
+      PYTHON_VERSION: 3.5.2
```
Comment (vam-google) [inline .circleci/config.yml]: 3.5 is generally ok, but our workstations on 3.7. Please consider changing this to 3.7 to minimize number of differences (monolyth gapic-generator CI still uses these versions, but it has to be updated as well, I just haven't a chance to make that small PR).

Comment (noahdietz) [reply .circleci/config.yml]: Done!

File: .circleci/config.yml (hunk 1 above)
Comment (noahdietz) [reply .circleci/config.yml]: Done!

**Pull request merged**